### PR TITLE
Search a list of strings using BM25

### PR DESCRIPTION
**Pull request by RobertHua96**

### CODE REVIEW

File: relevanceai/vector_tools/spar.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+from abc import abstractmethod
+import numpy as np
+import pandas as pd
+from rank_bm25 import *
```
Comment (boba-and-beer) [inline relevanceai/vector_tools/spar.py:4]: how do i install this

File: relevanceai/vector_tools/spar.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+from rank_bm25 import BM25Okapi
+from nltk.tokenize import word_tokenize
+from gensim.parsing.preprocessing import STOPWORDS
```
Comment (boba-and-beer) [inline relevanceai/vector_tools/spar.py:7]: how do i install this

File: relevanceai/vector_tools/spar.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            lst1.append(str_t)
+
+        return lst1
+
+    def search(self, query: str, k: int = 50):
+        tokenised_query = query.split(" ")
+        results = self.bm25.get_top_n(tokenised_query, self.corpus, n=k)
+        return results
+
+    """
```
Comment (boba-and-beer) [inline relevanceai/vector_tools/spar.py:48]: move to top of file

File: relevanceai/vector_tools/spar.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
+from abc import abstractmethod
```
Comment (boba-and-beer) [inline relevanceai/vector_tools/spar.py:1]: cool import bro where do u use it

File: relevanceai/vector_tools/spar.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            self.stopwords_removal_gensim_custom
+        )
+        self.corpus = self.df["cleaned_tokens"].tolist()
+        self.bm25 = BM25Okapi(self.corpus)
+
+    def spl_chars_removal(self, lst):
+        lst1 = list()
+        for element in lst:
+            str = ""
+            str = re.sub("[⁰-9a-zA-Z]”,” ", element)
```
Comment (boba-and-beer) [inline relevanceai/vector_tools/spar.py:26]: you're missing a `re` import this will fail

**Review by boba-and-beer — CHANGES_REQUESTED** [review]

File: relevanceai/vector_tools/spar.py (hunk 1 above)
Comment (RobertHua96) [reply relevanceai/vector_tools/spar.py:4]: I already slapped it to the requirements file

File: relevanceai/vector_tools/spar.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import re
+import numpy as np
+import pandas as pd
+from rank_bm25 import *
+from rank_bm25 import BM25Okapi
+from nltk.tokenize import word_tokenize
+from gensim.parsing.preprocessing import STOPWORDS
+
+"""
```
Comment (boba-and-beer) [inline relevanceai/vector_tools/spar.py:9]: Very top of file my friend. Before imports.

File: relevanceai/vector_tools/spar.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        for element in lst:
+            str = ""
+            str = re.sub("[⁰-9a-zA-Z]”,” ", element)
+            lst1.append(str)
+        return lst1
+
+    def stopwords_removal_gensim_custom(self, lst):
+        all_stopwords_gensim = STOPWORDS
+        lst1 = list()
+        for str in lst:
```
Comment (boba-and-beer) [inline relevanceai/vector_tools/spar.py:48]: Don't use `str` in a for loop. `str` is a very useful python default function. Use `string` or `x` as that might be better

**Review by boba-and-beer — CHANGES_REQUESTED** [review]

**Comment by boba-and-beer** [comment]
Closing for now as this will not be maintained in the future.

**Pull request closed without merging**